process.hpp

```diff
@@ -61,6 +61,8 @@ class Process {
   id_type get_id() const noexcept;
   ///Wait until process is finished, and return exit status.
   int get_exit_status() noexcept;
+  ///Check if the process has finished, return true and set the exit status otherwise return false
+  bool try_get_exit_status(int *exit_status) noexcept;
```

Comment:
Looks good, but why not: `bool try_get_exit_status(int &exit_status) noexcept;`?

Reply:
cpplint aka the google C++ guide suggests pointers for non-const aka out parameters. 

But as always, this is highly subjective. I happily change it to a reference if you like.

Reply:
I think google's C++ guide might be somewhat outdated, the core guidelines is better I think. But in this case, if it is a non-const pointer parameter, one might ask if we change the pointer, or if we change the value the pointer points to. With non-const reference parameter, we tell the users that we are changing the value. In other words, please use non-const reference parameter:)

Reply:
Sorry, my argument above is flawed since you pass the pointer by value:) The question is rather if the output parameter should be optional or not, that is if we should accept `nullptr` as argument.

Reply:
I changed it now to a reference. I am fine either way.